init.py

```diff
@@ -127,7 +128,32 @@ def gen_cycle_graph(target_dir, vertices):
     with open(os.path.join(target_dir, 'fullgraph_%s.txt'%(vertices)), 'a') as out_file:
         for i in range(0, vertices - 1):
             out_file.write('%s A %s \n'%(i, i + 1))
-        out_file.write('%s A %s \n'%(vertices - 1, 0))   
+        out_file.write('%s A %s \n'%(vertices - 1, 0))
+
+def gen_free_scale_graph(target_dir, n, k, labels, reverse_edges=False):
+    g = {
+        i: [(j, random.choice(labels))
+            for j in range(k)] for i in range(k)
+    }
```

Comment:
Suggestion: use a 3xn matrix to store the graph. In this way you can use the _to_file function. Moreover, the triples will be written to the file in the insertion order, whereas with the dictionary they won't.
I believe it's not gonna change much in the overall logic.

Reply:
Ok, I will change it. But I cant open source code of `_to_file` function (may be due to force-push). Now I want to merge this pull request and make refactoring which you suggested.